tests/test_sdist.py

```diff
@@ -60,3 +60,44 @@ def test_invalid_wheel_file_exception(mock_download_url, tmp_path: pathlib.Path)
     text_file.write_text("This is a test file")
     with pytest.raises(zipfile.BadZipFile):
         sdist._download_wheel_check(fake_dir, fake_url)
+
+
+@patch("fromager.sources.build_sdist")
+def test_sdist_build_exists(mock_build_sdist, tmp_path):
+    # Set any random package name
+    dist_name = "my-pkg"
+    version_string = "1.2"
+    expected_base = "mypkg-1.2.tar.gz"
+
+    # Create builds directory
+    sdists_repo = pathlib.Path(tmp_path)
+    builds = sdists_repo / "builds"
+    builds.mkdir()
+
+    # place the random package name in builds directory
+    archive = builds / expected_base
+    archive.write_text("not-empty")
+
+    req = Requirement(dist_name)
+    ver = Version(version_string)
+
+    mock_build_sdist.return_value = None
+
+    # Give call to find_sdist and check if sdist is present
+    find_sdist_result = finders.find_sdist(builds, req, ver)
```

Comment:
This is testing the finders package function, but we already have tests for those. 

I'm not sure how easy it's going to be to write a unit test for handle_requirement(), you might need to use an e2e test. For example, you could bootstrap something, then bootstrap the same thing again and look for a log message indicating that the existing sdist was used. Then you could remove the sdist and bootstrap again and look for a message indicating that it was created.

Keep in mind that the bootstrap log is appended, so you either need to provide a different file name each time or remove the file between bootstrap runs.

Reply:
Okay, will take a look